docs/src/architecture/notes.md

```diff
@@ -5,10 +5,60 @@ A note is a way of transferring assets between accounts. A note consists of a va
+In the above:
+* **Vault** is a set of assets that are stored in note.
+* **Script** that must be executed in a context of some account to claim the assets.
+* **Inputs** which are placed onto the stack before a note's script is executed.
+* **Serial number** is the note's identifier.
 
 A note's vault is basically the same as an account's vault. However, unlike an account, a note has a single executable script. This script is also a root of a [Miden program MAST](https://0xpolygonmiden.github.io/miden-vm/user_docs/assembly/main.html). A script is always executed in the context of a single account, and thus, may invoke account's functions. A note script can take parameters (passed via the stack) as inputs. A note script does not have to call any of account's functions. More generally, a note's script can call zero or more of an account's function. A note's serial number identifies the note and this is needed to create the note's hash and nullifier. 
+
+## Note hash
+The note hash is computed as: \
+
+`hash(hash(hash(hash(serial_num, [0; 4]), script_hash), input_hash), vault_hash)`
+
+This achieves the following properties:
+- Every note can be reduced to a single unique hash.
+- To compute a note's hash, we do not need to know the note's serial_num. Knowing the hash
+    of the serial_num (as well as script hash, input hash and note vault) is sufficient.
+- Moreover, we define `recipient` as:
+    `hash(hash(hash(serial_num, [0; 4]), script_hash), input_hash)`
+    This allows computing note hash from recipient and note vault.
+- We compute hash of serial_num as hash(serial_num, [0; 4]) to simplify processing within
+the VM.
+
+## Note nullifier
+The nullifier is the note's index in the [Nullifier DB](https://0xpolygonmiden.github.io/miden-base/architecture/state.html#nullifier-database). The Nullifier DB stores the information whether a note was already consumed. 
+
+The nullifier is computed as `hash(serial_num, script_hash, input_hash, vault_hash)`.
+This achieves the following properties:
+- Every note can be reduced to a single unique nullifier.
+- We cannot derive a note's hash from its nullifier.
+- To compute the nullifier we must know all components of the note: serial_num,
+script_hash, input_hash and vault hash.
+
+## Types of notes
+There are two types of notes in Miden. Notes can be stored privately in the Notes DB with only the note hash. Or notes can be stores publicly with all data.
+
+## Note lifecycle
+For a note to exist it must be present in the [Notes DB](https://0xpolygonmiden.github.io/miden-base/architecture/state.html#notes-database) kept by the Miden Node(s). New notes are being produced when executing a transaction. They can be produced locally by users in local transactions or by the operator in a network transaction.
+
+* If local transaction: 
+  * Alice produces a new note when executing and proving a transaction
+  * Alice sends proof and note hash with potential data to the Operator
```

Comment:
I think we should clarify that Alice sends a proof of transaction execution. This proof includes a list of hashes for all newly created notes.

Also, "potential data" sounds a bit weird here. Are we trying to say that sending the actual data is optional?